mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -229,6 +229,48 @@ static Optional<Type> convertVecType(sycl::VecType type,
   return convertBodyType("class.sycl::_V1::vec", type.getBody(), converter);
 }
 
+int convertToLLVMAddressSpace(mlir::sycl::AccessAddrSpace AddrSpace) {
+  switch (AddrSpace) {
+  case AccessAddrSpace::Private:
+    return 0;
+  case AccessAddrSpace::Global:
+    return 1;
+  case AccessAddrSpace::Constant:
+    return 2;
+  case AccessAddrSpace::Local:
+    return 3;
+  case AccessAddrSpace::ExtIntelGlobalDevice:
+    return 4;
+  case AccessAddrSpace::ExtIntelHost:
+    return 5;
+  case AccessAddrSpace::Generic:
+    return 6;
+  default:
+    llvm_unreachable("Unknown address mode");
+  }
+}
+
+/// Converts SYCL atomic type to LLVM type.
+static Optional<Type> convertAtomicType(sycl::AtomicType type,
+                                        LLVMTypeConverter &converter) {
+  // return convertBodyType("class.sycl::_V1::atomic", type.getBody(),
+  // converter);
+  auto convertedTy = LLVM::LLVMStructType::getIdentified(
+      &converter.getContext(), "class.sycl::_V1::atomic");
+  auto elementType = LLVM::LLVMPointerType::get(
+      type.getDataType(), convertToLLVMAddressSpace(type.getAddrSpace()));
```

Comment:
Do we need `convertToLLVMAddressSpace`? Can we do something like:
```suggestion
      type.getDataType(), (int)(type.getAddrSpace());
```